src/main/kotlin/StringAddCalculator.kt

```diff
@@ -0,0 +1,54 @@
+import java.lang.RuntimeException
+
+class StringAddCalculator(strExpression: String?) {
+    var result: Int = 0
+    var arrNumber: List<String> = ArrayList()
+
+    init {
+        if (strExpression.isNullOrEmpty())
+            result = 0
+        else {
+            checkCustomDelimiter(strExpression)
+        }
+    }
+
+    fun checkCustomDelimiter(strExpression: String) {
+        val check = Regex("//(.)\n(.*)").find(strExpression)
+        if (check != null) {
+            check.let {
+                val customDelimiter = it.groupValues[1]
+                arrNumber = it.groupValues[2].split(customDelimiter)
+                amountArrNumber()
+            }
+        } else {
+            splitCommaAndColon(strExpression)
+        }
+    }
+
+    fun splitCommaAndColon(strExpression: String) {
+        arrNumber = strExpression.split(",|:".toRegex())
+        strExpressionSizeCheck(arrNumber)
+    }
+
+    fun strExpressionSizeCheck(arrNumber: List<String>) {
+        if (arrNumber.size == 1) {
+            result = checkNagative(arrNumber.get(0).toInt())
```

Comment:
Kotlin에서는 List의 각 원소에 접근할 때, 꼭 `get()` 함수를 쓰지 않고 아래와 같이 `[index]`로 접근할 수 있어요! 🙂
```suggestion
            result = checkNagative(arrNumber[0].toInt())
```

그리고 List의 첫 번째 원소에 대해서는 `first()` 함수를 이용해볼 수도 있을 것 같아요. 😉
```suggestion
            result = checkNagative(arrNumber.first().toInt())
```

이 외에도 Kotlin에서는 컬렉션에 다양하고 또 유용한 확장 함수들을 많이 제공하고 있으니 학습 테스트를 작성해보며 학습해보시면 좋을 것 같아요. 💪